src/test/java/org/isf/medicalsinventory/Tests.java

```diff
@@ -811,4 +830,94 @@ void testValidateMedicalInventory() throws Exception {
+			MedicalInventoryRow medicalInventoryRowThree = testMedicalInventoryRow.setup(inventory, medical, lotThree, false);
+			medicalInventoryRowThree.setId(3);
+			medicalInventoryRowIoOperationRepository.saveAndFlush(medicalInventoryRowOne);
+			medicalInventoryRowIoOperationRepository.saveAndFlush(medicalInventoryRowTwo);
+			medicalInventoryRowIoOperationRepository.saveAndFlush(medicalInventoryRowThree);
+			firstMovement = medicalStockIoOperation.newMovement(firstMovement);
+			medicalStockWardIoOperation.newMovementWard(wardMovement);
+			medicalStockIoOperationRepository.saveAndFlush(firstmedicalStock);
+			movementWardIoOperationRepository.saveAndFlush(wardMovement);
+			int inventoryId = inventory.getId();
+			List<MedicalInventoryRow> medicalInventoryRows = medicalInventoryRowManager.getMedicalInventoryRowByInventoryId(inventoryId);
+			assertThat(medicalInventoryRows).isNotEmpty();
+			assertThat(medicalInventoryRows).hasSize(3);
+			firstMovement.setQuantity(100);
+			firstMovement.setDate(LocalDateTime.now());
+			firstmedicalStock = testMedicalStock.setup(firstMovement);
+			medicalStockIoOperation.newMovement(firstMovement);
+			medicalStockIoOperationRepository.saveAndFlush(firstmedicalStock);
+			Lot lotfour = testLot.setup(medical, false);
+			lotfour.setCode("LOT-004");
+			lotfour = lotIoOperationRepository.save(lotfour);
+			Movement secondMovement = testMovement.setup(medical, dischargeType, ward, lotfour, null, false);
+			secondMovement.setQuantity(50);
+			secondMovement.setDate(LocalDateTime.now());
+			MedicalStock secondMedicalStock = testMedicalStock.setup(secondMovement);
+			medicalStockIoOperation.newMovement(secondMovement);
+			medicalStockIoOperationRepository.saveAndFlush(secondMedicalStock);
+			Medical secondMedical = testMedical.setup(medicalType,false);
+			secondMedical.setProdCode("TP2");
+			secondMedical.setDescription("test description");
+			secondMedical = medicalsIoOperationRepository.save(secondMedical);
+			Lot lotFive = testLot.setup(secondMedical, false);
+			lotFive.setCode("LOT-005");
+			lotFive = lotIoOperationRepository.save(lotFive);
+			initialMovement = testMovement.setup(secondMedical, chargeType, ward, lotFive, supplier, false);
+			initialMovement.setQuantity(200);
+			initialMedicalStock = testMedicalStock.setup(initialMovement);
+			medicalStockIoOperation.newMovement(initialMovement);
+			medicalStockIoOperationRepository.saveAndFlush(initialMedicalStock);
+			Movement thirdMovement = testMovement.setup(secondMedical, dischargeType, ward, lotFive, null, false);
+			thirdMovement.setDate(LocalDateTime.now());
+			MedicalStock thirdMedicalStock = testMedicalStock.setup(thirdMovement);
+			medicalStockIoOperation.newMovement(thirdMovement);
+			medicalStockIoOperationRepository.saveAndFlush(thirdMedicalStock);
+			medicalInventoryManager.validateMedicalWardInventoryRow(inventory, medicalInventoryRows);
+		}).isInstanceOf(OHDataValidationException.class);
```

Comment:
I think here you should differentiate by the content of the exception:
![image](https://github.com/user-attachments/assets/73846a56-bd4b-4651-95ef-aa02bc132e04)
 
You can make one test for each case and then assert the number of messages included in the returned `errors` variable:
![image](https://github.com/user-attachments/assets/4b6d6d5c-8415-453f-b330-7bc56cdc1794)

I think the best is to put some comments in the tests construction to better clarify what you are doing and to test the different errors in separated tests (as already suggested) by factorizing the common parts